### PR TITLE
Add region interactor, improve selection handling.

### PR DESCRIPTION
**Pull request by jheer**
- Add `region` interactor for 2D selection of individual point values.
- Add SQL `isIn` function.
- Add use of SQL IN operator for single-value point selections.
- Add `channels` mark property to spec types.
- Fix empty selection resolution to use false literal.
- Fix interactor facet handling and datum extraction.
- Refactor BrushStyles types to dedicated file.
- Update documentation and example specs.

### CODE REVIEW

File: packages/spec/src/spec/marks/Marks.ts [hunk 1]
```diff
@@ -432,6 +432,12 @@ export interface MarkOptions {
     | (TipOptions & { pointer?: TipPointer })
     | ParamRef;
 
+  /**
+   * Additional named channels, for example to include in a tooltop.
```
Comment (domoritz) [inline packages/spec/src/spec/marks/Marks.ts]:
```suggestion
   * Additional named channels, for example to include in a tooltip.
```

File: packages/plot/src/interactors/util/intersect.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function intersect(svg, el, extent) {
       list.push(child);
     }
   }
-  return list;
+  return /** @type {SVGElement[]} */(list);
```
Comment (domoritz) [inline packages/plot/src/interactors/util/intersect.js]: Might be easier/clearer to move this up to line 18?

**Review by domoritz — APPROVED** [review]
Looks great.

Screenshot 2024-10-30 at 13 43 36

File: packages/spec/src/spec/marks/Marks.ts (hunk 1 above)
Comment (jheer) [reply packages/spec/src/spec/marks/Marks.ts]: Hahaha. Good catch.

**Pull request merged**